internal/fuzz/ratelimit.go

```diff
@@ -0,0 +1,130 @@
+package fuzz
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+)
+
+// PerformRateLimitFuzz performs rate limit detection on target URLs within a specified timespan
+func PerformRateLimitFuzz(ctx context.Context, inputTargets []string, config *webscan.FuzzRateLimitConfig) (*webscan.FuzzRateLimitReport, error) {
+	report := &webscan.FuzzRateLimitReport{Config: config}
+
+	client := &http.Client{
+		Timeout: time.Duration(config.Timeout) * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Since we don't care if its insecure or not

> This setting disables the verification of the server's certificate. It means the client won't check if the server's certificate is valid (e.g., issued by a trusted Certificate Authority (CA), matches the hostname, or has not expired).